src/server/pkg/worker/master.go

```diff
@@ -261,14 +265,20 @@ nextInput:
 
 // jobManager feeds datums to jobs
 func (a *APIServer) runJob(ctx context.Context, jobInfo *pps.JobInfo, pool *grpcutil.Pool) error {
+	jobStart := time.Now()
+	defer func() { protolion.Infof("running job (%v) took (%v)\n", jobInfo, time.Since(jobStart)) }()
+	protolion.Infof("running job (%v)\n", jobInfo)
 	pfsClient := a.pachClient.PfsAPIClient
 	objectClient := a.pachClient.ObjectAPIClient
 	ppsClient := a.pachClient.PpsAPIClient
 
 	jobID := jobInfo.Job.ID
 	var jobStopped bool
 	var jobStoppedMutex sync.Mutex
+	retryCount := 0
```

Comment:
It doesn't get logged, although we log when the job gets restarted, so I think just adding the restart count in that log line would work well. You have the `JobInfo` available to you in that scope too so it should be pretty easy.